File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/loginfo/reader/LogcatReader.java
package com.didichuxing.doraemonkit.kit.loginfo.reader;

import java.io.IOException;
import java.util.List;

public interface LogcatReader {

    /**
     * Read a single log line, ala BufferedReader.readLine().
     *
     * @return
     * @throws IOException
     */
    String readLine() throws IOException;

    /**
     * Kill the reader and close all resources without throwing any exceptions.
     */
    void killQuietly();

    boolean readyToRecord();

    List<Process> getProcesses();

}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/loginfo/LogcatReader.kt
package com.didichuxing.doraemonkit.kit.loginfo

import java.io.IOException

/**
 * @author lostjobs created on 2020/6/28
 */
interface LogcatReader {
    @Throws(IOException::class)
    fun readLine(): String?

    fun killQuietly()

    fun readyToRecord(): Boolean

    fun getProcesses(): List<Process>
}